src/main/java/crawlercommons/sitemaps/SiteMapParser.java

```diff
@@ -595,6 +595,9 @@ protected AbstractSiteMap processXml(URL sitemapUrl, InputSource is) throws Unkn
             factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
             factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
             factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+            if (!"true".equalsIgnoreCase(System.getProperty("crawler-commons.sitemap.allowDocTypes"))) {
```

Comment:
[Boolean.getBoolean(...)](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Boolean.html#getBoolean(java.lang.String)) would be shorter and does exactly the same.

Until know properties are only used in the class SiteMapTester. To keep the SiteMapParser consistent: what about adding a setter method `setAllowDocTypeDefinitions(boolean)` which is used in SiteMapTester and unit tests. This would allow to switch the support for DTDs on and off programmatically.